File: verification/verify_frontend.py
from playwright.sync_api import sync_playwright, expect
import time

def verify_homepage():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        # Navigate to the homepage
        page.goto("http://localhost:3000")

        # Check if the title is correct
        expect(page).to_have_title("Осмотр ТС - Сентрас Иншуранс")

        # Check if the main elements are visible
        expect(page.locator("h1")).to_have_text("Осмотр Транспортного Средства")

        # Take a screenshot of the homepage
        page.screenshot(path="verification/homepage.png")

        browser.close()

def verify_video_page():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        # Navigate to the video page
        page.goto("http://localhost:3000/video.html")

        # Check if the title is correct
        expect(page).to_have_title("Видео-осмотр - Сентрас Иншуранс")

        # Take a screenshot of the video page
        page.screenshot(path="verification/video_page.png")

        browser.close()

if __name__ == "__main__":
    print("Verifying homepage...")
    verify_homepage()
    print("Verifying video page...")
    verify_video_page()

File: verification/verify_frontend.js
const { chromium } = require('playwright');
const path = require('path');

(async () => {
    // We need to launch with options to mock camera/microphone
    const browser = await chromium.launch({
        headless: true, // headless must be true for verification instruction
        args: [
            '--use-fake-ui-for-media-stream',
            '--use-fake-device-for-media-stream',
        ]
    });
    const context = await browser.newContext({
        permissions: ['camera', 'microphone']
    });
    const page = await context.newPage();

    console.log("Navigating to Live Inspection page...");
    await page.goto('http://localhost:3000/live.html');

    // Wait for the start button
    await page.waitForSelector('#start-btn');
    console.log("Start button found.");

    // Click Start
    await page.click('#start-btn');
    console.log("Clicked Start.");

    // The "Status" should change to "Connecting..." or "Listening"
    // Since we don't have a real Gemini backend connected with key, it might error out or get stuck at "Connecting" or fail.
    // However, the test is to verify the UI *can* be interacted with and the Stop button logic *can* be triggered.
    // If the websocket connection fails immediately (no key), we might see an error.

    // Let's see what happens.
    // We wait a bit.
    await page.waitForTimeout(2000);

    // Take a screenshot of the active state
    await page.screenshot({ path: 'verification/live_active.png' });
    console.log("Active state screenshot taken.");

    // Check if Stop button is enabled.
    const stopBtn = await page.$('#stop-btn');
    const isEnabled = await stopBtn.isEnabled();
    console.log(`Stop button enabled: ${isEnabled}`);

    if (isEnabled) {
        // Click Stop
        await page.click('#stop-btn');
        console.log("Clicked Stop.");

        // Verify button becomes disabled immediately (Frontend logic check)
        const isDisabledAfter = await stopBtn.isDisabled();
        console.log(`Stop button disabled after click: ${isDisabledAfter}`);

        if (!isDisabledAfter) {
            console.error("FAILURE: Stop button was not disabled immediately.");
        }

        // Take a screenshot of the stopped state
        await page.screenshot({ path: 'verification/live_stopped.png' });
    } else {
        console.log("Stop button was not enabled (likely connection failed due to missing API key), which is expected in this env without key.");
        // Take a screenshot of the error state
        await page.screenshot({ path: 'verification/live_error.png' });
    }

    await browser.close();
})();
